tests/test_model_chat.py

```diff
@@ -24,6 +27,20 @@ def test_model_name(model_path, expected_name):
     assert chatbot.model_name == expected_name
 
 
+def test_retriever_is_called_when_present():
+    retriever = MagicMock()
+    chatbot = ConsoleChatBot(
+        model="/var/model/file", client=None, retriever=retriever, loaded={}
+    )
+    assert chatbot.retriever == retriever
+    user_query = "test"
+    try:
+        chatbot.start_prompt(content=user_query, logger=logger)
+    except Exception as e:  # pylint: disable=broad-exception-caught
+        logger.info(e)
+        retriever.augmented_context.assert_called_with(user_query=user_query)
```

Comment:
Sorta nitpicky... Is it possible to use the `pytest.raises` function instead to ensure the proper exception class is being raised? For example...

```suggestion
    with pytest.raises(ChatException) as exc_info:
        chatbot.start_prompt(content=user_query, logger=logger)
        logger.info(exc_info)
        retriever.augmented_context.assert_called_with(user_query=user_query)
```